example/src/MutableReference.hs

```diff
@@ -71,20 +69,15 @@ instance Eq1 v => Eq (Ref v) where
 instance Show1 v => Show (Ref v) where
   show (Ref v) = showsPrec1 10 v ""
 
-instance Show (Internal Action) where
-  show (Internal New           sym) = "New (" ++ show sym ++ ")"
-  show (Internal (Read  ref)   sym) =
-    "Read ("  ++ show ref ++ ") (" ++ show sym ++ ")"
-  show (Internal (Write ref i) sym) =
-    "Write (" ++ show ref ++ ") (" ++ show i ++ ") (" ++ show sym ++ ")"
-  show (Internal (Inc   ref)   sym) =
-    "Inc ("   ++ show ref ++ ") (" ++ show sym ++ ")"
-
 instance ShowAction Action where
-  showAction New           = ShowResponse "New"                                   show
-  showAction (Read ref)    = ShowResponse ("Read " ++ show ref)                   show
-  showAction (Write ref i) = ShowResponse ("Write " ++ show ref ++ " " ++ show i) show
-  showAction (Inc ref)     = ShowResponse ("Inc " ++ show ref)                    show
+  showAction New           =
+    ShowResponse "New"                                             show
```

Comment:
Maybe we should add `showResponse` smart constructor that always uses `show` for the response.